datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -1363,6 +1380,57 @@ fn get_filter_column(
     filter_columns
 }
 
+// Get buffered data sliece by specific batch index and for specified column indices only
```

Comment:
I think `buffered_indices` are row indices (not column indices), which confused me for a moment

Maybe this would be clearer:

```suggestion
/// Get `buffered_indices` rows for `buffered_data[buffered_batch_idx]`
```